spec/outputs/rabbitmq_spec.rb

```diff
@@ -69,4 +69,64 @@
     end
 
   end
+
+  # TODO: Redo all this
+  # This is a crazy test to fix an urgent bug. It tests that we actually to change the
+  # store exchange value when an exception happens
+  # This should be a small number of lines, but until we refactor we must jump through hoops
+  # to put the plugin in the proper state and extract the proper state to test
+  describe "retrying a publish" do
+    let(:settings) {
+      {
+        "host" => "localhost",
+        "exchange_type" => "topic",
+        "exchange" => "foo",
+        "key" => "%{foo}"
+      }
+    }
+    let(:event) { LogStash::Event.new("foo" => "bar")}
+
+    subject { LogStash::Outputs::RabbitMQ.new(settings) }
+
+    before do
+      allow(subject).to receive(:connect) {
+                          subject.instance_variable_get(:@connected).set(true)
+                        }
+
+      exchange_invocation = 0
+
+      @most_recent_exchange = nil
```

Comment:
We need them because it's autable value, not suitable for a let. Keep in mind, I'm planning on refactoring away the need for this setup by tomorrow.